src/main/java/io/supertokens/webserver/api/core/HelloAPI.java

```diff
@@ -76,9 +74,14 @@ protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws IO
     }
 
     private void handleRequest(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        // API is app specific
+        // TODO add rate limiter to this API based on appIdentifier
         try {
-            TenantIdentifierWithStorage tenantIdentifierWithStorage =  getTenantIdentifierWithStorageFromRequest(req);
-            tenantIdentifierWithStorage.getStorage().getKeyValue(tenantIdentifierWithStorage, "Test");
+            AppIdentifierWithStorage appIdentifierWithStorage =  getAppIdentifierWithStorage(req);
+
+            for (Storage storage : appIdentifierWithStorage.getStorages()) {
+                storage.getKeyValue(appIdentifierWithStorage.getAsPublicTenantIdentifier(), "Test");
```

Comment:
add a comment why using public tenant is ok

Reply:
Done